osu.Game/Online/API/APIAccess.cs

```diff
@@ -540,6 +574,11 @@ public enum APIState
         /// </summary>
         Failing,
 
+        /// <summary>
+        /// Waiting on second factor authentication.
+        /// </summary>
+        RequiresSecondFactorAuth,
```

Comment:
Adding this to `APIState` is going to require a sweep of all `APIAccess` checks we have and seeing if they're broken by this or not.

A few places I could see being broken at a first glance:

https://github.com/ppy/osu/blob/87a76b9ee80055a85cb51a12e0a70083c1ba63e2/osu.Game/Online/OnlineViewContainer.cs#L63-L87
https://github.com/ppy/osu/blob/87a76b9ee80055a85cb51a12e0a70083c1ba63e2/osu.Game/Overlays/Comments/CommentEditor.cs#L181
https://github.com/ppy/osu/blob/87a76b9ee80055a85cb51a12e0a70083c1ba63e2/osu.Game/Overlays/AccountCreationOverlay.cs#L112-L128

and maybe things that use

https://github.com/ppy/osu/blob/87a76b9ee80055a85cb51a12e0a70083c1ba63e2/osu.Game/Online/API/APIAccess.cs#L461

Reply:
I think most of these are okay. Anything which was already valid in "connecting" should be valid in the 2FA state too. I'll fix the two switch based ones (which really should have throw-on-missing).